File: src/network/PacketAndIP.java
package network;

import java.net.InetAddress;

import network.Packet;

public class PacketAndIP
{	
	private Packet packet;
	private InetAddress ip;

	public PacketAndIP(Packet packet, InetAddress ip)
	{
		this.packet = packet;
		this.ip = ip;
	}

	// Getter
	public Packet getPacket() { return packet; }
	public InetAddress getIP() { return ip; }
}

File: cpp/src/network/PacketAndIP.cpp
#include "PacketAndIP.hpp"

PacketAndIP::PacketAndIP(Packet* packet, const sf::IpAddress& ipArg) : packet(packet)
{
	ip = ipArg;
}

PacketAndIP::~PacketAndIP()
{} // should not delete packet

Packet* PacketAndIP::getPacket()
{
	return packet;
}

const sf::IpAddress& PacketAndIP::getIP()
{
	return ip;
}
